### PR TITLE
Add Adjacent and Neighbor functions

### PR DESCRIPTION
**Pull request by JacobHearst**
Hello, I've added the ability to calculate the adjacent block of a geohash as well as a function to get all the neighboring blocks

### CODE REVIEW

File: Sources/Geohash/Extensions.swift [hunk 1]
```diff
@@ -0,0 +1,25 @@
+//
+//  File.swift
+//  
+//
+//  Created by Jacob Hearst on 10/31/21.
+//
+
+import Foundation
+
+extension StringProtocol {
```
Comment (nh7a) [inline Sources/Geohash/Extensions.swift]: Can you make all those extensions private and move to `Geohash.swift`? I'd like to maintain `Geohash` in one file as long as it makes sense.

File: Sources/Geohash/Geohash.swift [hunk 2]
```diff
@@ -31,7 +31,7 @@ public enum Geohash {
             .joined(separator: "")
         guard bits.count % 5 == 0 else { return nil }
         // bits = 1101000100101011011111010111100110010110101101101110001
-
+
```
Comment (nh7a) [inline Sources/Geohash/Geohash.swift]: Please remove all those unnecessary spaces.

File: Sources/Geohash/Geohash.swift [hunk 3]
```diff
@@ -159,14 +159,79 @@ public extension CLLocationCoordinate2D {
...
+            case .w:
+                return ["0145hjnp", "028b"]
+            }
+        }
+    }
+
+    static func adjacent(geohash: String, direction: Direction) throws -> String {
+        let lastChar = geohash.last!
+        var parent = String(geohash.dropLast())
+        let type = geohash.count % 2;
```
Comment (nh7a) [inline Sources/Geohash/Geohash.swift]:
```suggestion
        let type = geohash.count % 2
```

File: Sources/Geohash/Geohash.swift [hunk 4]
```diff
@@ -159,14 +159,79 @@ public extension CLLocationCoordinate2D {
...
+
+        // Check for edge-cases which don"t share common prefix
+        if direction.border[type].contains(lastChar) && !parent.isEmpty {
```
Comment (nh7a) [inline Sources/Geohash/Geohash.swift]:
```suggestion
        if direction.border[type].contains(lastChar), !parent.isEmpty {
```

File: Sources/Geohash/Geohash.swift [hunk 5]
```diff
@@ -159,14 +159,79 @@ public extension CLLocationCoordinate2D {
...
+            }
+        }
+    }
+
+    static func adjacent(geohash: String, direction: Direction) throws -> String {
+        let lastChar = geohash.last!
+        var parent = String(geohash.dropLast())
+        let type = geohash.count % 2;
+
+        // Check for edge-cases which don"t share common prefix
```
Comment (nh7a) [inline Sources/Geohash/Geohash.swift]:
```suggestion
        // Check for edge-cases which don't share common prefix
```

**Review by nh7a — COMMENTED** [review]
Cool. Can you also add test cases?

**Review by nh7a — APPROVED** [review]
Cool.
